moto/transfer/types.py

```diff
@@ -0,0 +1,185 @@
+from dataclasses import asdict, dataclass, field
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Literal, Optional, TypedDict
+
+from moto.core.common_models import BaseModel
+
+
+class UserHomeDirectoryType(str, Enum):
+    PATH = "PATH"
+    LOGICAL = "LOGICAL"
+
+
+class UserHomeDirectoryMappingType(str, Enum):
+    FILE = "FILE"
+    DIRECTORY = "DIRECTORY"
+
+
+class UserHomeDirectoryMapping(TypedDict):
+    Entry: str
```

Comment:
These properties (from all the TypedDict's _and_ from @dataclasses) should be snake-cased, to be in-line with the rest of the code base. So `entry`, `target`, etc. 

`type` is a keyword, so our linter may complain if you use that as variable name, but that can just become something like `mapping_type`.